plugins/modules/net_tools/ldap/ldap_entry.py

```diff
@@ -158,12 +167,29 @@ def _add():
         return action
 
     def delete(self):
-        """ If self.dn exists, returns a callable that will delete it. """
+        """ If self.dn exists, returns a callable that will delete either
+        the item itself if the recursive option is not set or the whole branch
+        if it is. """
         def _delete():
             self.connection.delete_s(self.dn)
 
+        def _delete_recursive():
+            """ Attempt recurive deletion using the subtree-delete control.
+            If that fails, do it manually. """
+            try:
+                subtree_delete = ldap.controls.ValueLessRequestControl('1.2.840.113556.1.4.805')
+                self.connection.delete_ext_s(self.dn, serverctrls=[subtree_delete])
```

Comment:
Hmm, according to https://ldapwiki.com/wiki/Subtree%20Delete%20Control this is supported only by very few LDAP servers. I guess it needs to be documented that this only works in very specific situations.

Reply:
Ah sorry, I somehow completely missed the fallback...